src/pvt.c

```diff
@@ -278,6 +283,139 @@ static u8 filter_solution(gnss_solution* soln, dops_t* dops)
   return 0;
 }
 
+static bool chi_test(u8 n_used, double omp[n_used], double rx_state[],
+                     double *residual)
```

Comment:
Maybe consolidate with the other `chi_test`, you could apply the clock correction outside the function